src/Raven.Server/Documents/Handlers/BatchRequestParser.cs

```diff
@@ -161,8 +162,33 @@ private static void ReturnBuffer(CommandData[] cmds)
                         positionInListToCommandIndex, 
                         cmds);
                 }
+
+                command.ParsedCommands = new ArraySegment<CommandData>(cmds, 0, index + 1);
+                return await IsClusterTransaction(stream, parser, buffer, state);
+            }
+        }
+
+        private static async Task<bool> IsClusterTransaction(Stream stream, UnmanagedJsonParser parser, JsonOperationContext.ManagedPinnedBuffer buffer, JsonParserState state)
+        {
+            while (parser.Read() == false)
+                await RefillParserBuffer(stream, buffer, parser);
+
+            if (ReadClusterTransactionProperty(state))
+            {
+                while (parser.Read() == false)
+                    await RefillParserBuffer(stream, buffer, parser);
+
+                return state.CurrentTokenType == JsonParserToken.True;
             }
-            return new ArraySegment<CommandData>(cmds, 0, index + 1);
+
+            return false;
+        }
+
+        private static unsafe bool ReadClusterTransactionProperty(JsonParserState state)
+        {
+            return state.CurrentTokenType == JsonParserToken.String &&
+                   GetLongFromStringBuffer(state) == 6085037597358648387 && // ClusterT
```

Comment:
You are not checking the size of the string here, can cause invalid memory access